### DISCUSSION TITLE
UV Splitting Issue When Importing USD Files into Maya

### DISCUSSION THREAD

**Discussion by GabrielCGI** [Autodesk/maya-usd#3882] [General]
hello,
![image](https://github.com/user-attachments/assets/2fec27b7-4ffc-4127-a2a4-3fdd6fe09a49)

When exporting a USD mesh from houdini and using the Maya Importer the object's UV are "splitted" per face.
Alembic works fine.

Is there an option somewhere to tick ?
Thanks !

**Comment by santosd** [comment]
Is this also happening with a different file format like FBX or OBJ or only happening with USD files?

**Reply by GabrielCGI** [reply]
thanks for the reply. it's append only with USD.

**Reply by cescocm** [reply]
Hello this is also happening with meshes coming from Blender. Are there news about this?

**Comment by 356Fab** [comment]
Is the maya usd importer not handling face corner uvs correctly?

**Comment by wallworm** [comment]
The USD Import does preserve the data as it comes in from USD. If the incoming USD has split UVS, they will come in split; if the USD data is sharing UV verts between faces, the UVs will not be split. I suspect that the original data in the other DCCs is either split or that the exporters in those other DCCs is splitting the data.

**Comment by 356Fab** [comment]
If that's the case, why does Blender import USD files exported from Houdini without issues? For example, when I export a test mesh like the rubberToy from Houdini as USD and import it into Maya (using the latest v0.31 USD plugin), the UVs appear split and disorganized—a checker texture becomes scrambled. Given that Blender and Houdini seamlessly exchange USD files, this suggests the problem lies specifically with Maya's USD importer. It appears either the importer is malfunctioning or mishandling UV data during translation.

**Reply by wallworm** [reply]
The files I've tested export/import as expected. Perhaps there is some specific mesh setting that we are not handling gracefully. Can you send us a sample of the USD file that is having issues for you?

**Reply by 356Fab** [reply]
Its an usdc export of the rubberToy test mesh from Houdini 
[rubberToy.zip](https://github.com/user-attachments/files/20416257/rubberToy.zip)

**Reply by wallworm** [reply]
Thank you for the sample. My suspicion is correct in that it's a mesh setting we don't yet handle correctly. This mesh is using leftHanded face orientation, and the importer is not properly translating all the primvars. I've lined up the fix for a future release. Related threads:

- https://github.com/Autodesk/maya-usd/issues/3690
- https://github.com/Autodesk/maya-usd/issues/3631

**Reply by cescocm** [reply]
Just a small clarification. Aren't these different issues? One is the data structure coming from Blender and the other one maya_usd importer not handling correclty leftHanded face orientation? Or are all part of the same issue?
Thanks!

**Reply by wallworm** [reply]
@cescocm My current understanding is simply that Maya USD isn't handling leftHanded face orientation at the moment. Please send me a file from Blender that is causing a problem and I can validated if there is another issue to address.

**Reply by cescocm** [reply]
Hi, here you can find a USD exported from Maya with 2 cubes one coming originally from Blender and one coming from Maya. You can see the differences in the data structure.
[testcube-v03.zip](https://github.com/user-attachments/files/20491533/testcube-v03.zip)

**Reply by wallworm** [reply]
> Hi, here you can find a USD exported from Maya with 2 cubes one coming originally from Blender and one coming from Maya. 

It's not going to do much good to have exports coming from MayaUSD. We need to see the data before it gets into Maya to properly evaluate. Can you share with us the USD you created from Blender?

**Reply by cescocm** [reply]
Here you can find the same cube directly from Blender with no Maya interaction.

https://www.dropbox.com/scl/fi/kygxc7alqvb0vralqyh4o/testcube-v01.usdc?rlkey=ja8xgipxc93y9454dtfl9uody&st=t0lleyd1&dl=0

**Reply by wallworm** [reply]
The USD file you sent from Blender doesn't have welded UVs in the USD data. That is the cause of the Blender file UVs not being welded. Without sharing UV (indices) at corners, you can expect that the UVs are not shared between adjacent faces. To fix, you should either weld the UVs in Blender before export, or weld them once imported.

**Reply by wallworm** [reply]
I also did a little more digging into the samples above, and _both_ the Houdini and Blender USD files seem to be missing st:indices that would indicate shared/welded UV verts. If the USD does contain shared UVs, then the UVs between faces won't be split. (The Houdini example above is leftHanded, but it turns out that is a separate issue that we do need to address).

**Reply by cescocm** [reply]
So, then we are talking about 2 different issues here:
**1.- UVs Not merged.**
The cube from Blender has "merged/welded/shared" UVs in Blender. Blender stores all facecorner data without deduplication and that is the data that is written into the usd from Blender.

**2.- leftHanded mesh setting not being handle correctly by maya_USD importer.**
If I understood correctly, this will be fixed by maya_USD eventually. In the meantime rightHanded works fine.

Thanks!

**Reply by wallworm** [reply]
Yes, that is true overall. I can't say how Blender itself stores data internally, but at least within the confines of the supplied USD file exported from Blender above, Maya is importing the USD UVs as supplied in the USD data.
